### ISSUE TITLE
Make a streamlined abstraction for Dropout and BatchNormalization

### ISSUE THREAD

**Issue by vzhong** [vzhong/pystacks#3]
In particular, these layers behave differently during training and testing. To do with this, we need to either add in a symbolic switch (eg. `ifelse`), or compile 2 different graphs. The latter seems like it would create a mess of the interfaces.
